lib/options.go

```diff
@@ -500,11 +500,38 @@ func (o Options) ForEachSpecified(structTag string, callback func(key string, va
 	for i := 0; i < structType.NumField(); i++ {
 		fieldType := structType.Field(i)
 		fieldVal := structVal.Field(i)
+		value := fieldVal.Interface()
 
 		shouldCall := false
 		switch fieldType.Type.Kind() {
 		case reflect.Struct:
+			// Unpack any guregu/null values
 			shouldCall = fieldVal.FieldByName("Valid").Bool()
+			valOrZero := fieldVal.MethodByName("ValueOrZero")
+			if shouldCall {
+				if valOrZero.IsValid() {
+					val := valOrZero.Call([]reflect.Value{})[0]
+					switch val.Kind() {
+					case reflect.Bool:
+						value = val.Bool()
+					case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+						value = val.Int()
+					case reflect.Uint, reflect.Uintptr, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+						value = val.Uint()
+					case reflect.Float32, reflect.Float64:
+						value = val.Float()
+					case reflect.String:
+						value = val.String()
+					default:
+						value = val.Interface()
+					}
+				} else {
```

Comment:
I think you can simply implement the `ValueOrZero` method for our own custom `types.Duration` type.